src/movies.js

```diff
@@ -1,25 +1,129 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+
+function getAllDirectors(moviesArray) {
+    let directorsArray = [];
+    directorsArray = moviesArray.map((element) => {
+        return element.director;
+    })
+    return directorsArray;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+
+
+function howManyMovies(moviesArray) {
+    let spielbergDramaMovies = moviesArray.filter((movie) =>
+        movie.director === "Steven Spielberg" && movie.genre.includes("Drama")
+        );
+        return spielbergDramaMovies.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+
+
+function scoresAverage(moviesArray) {
+    // if empty array
+    if(moviesArray.length === 0){
+        return 0;
+    }
+    //declaring + initializing the totalavg variable
+    let avgTotal = 0;
+    /*if(moviesArray.score = 0) {
+        continue;
+    }*/
+
+    let imdbTotalScores = moviesArray.reduce(function(accumulator, currentValue) {
+        if (currentValue.score) {
+            return accumulator + currentValue.score;
+        } else {
+            return accumulator;
+        }
+    }, 0)
+
+    let averageScores = imdbTotalScores/moviesArray.length
+
+    averageScores = Math.round(averageScores*100)/100;
+
+    return averageScores;
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+
+
+function dramaMoviesScore(moviesArray) {
+
+    let dramaMovies = moviesArray.filter((movie) =>
+        movie.genre.includes("Drama"));
+
+    if (dramaMovies.length===0) {
+        return 0;
+    }
+
+
+    let dramaMoviesTotalScores = dramaMovies.reduce((accumulator, currentValue) => {
```

Comment:
You previously wrote a function that calculates the average `scoresAverage`, so, instead of writing again the whole logic to do the average inside this `dramaMoviesScore` function, you should reuse `scoresAverage`.
After you `filter` the movies, you should just `return scoresAverage(dramaMovies)`.